<Type Name="IotHubClientErrorCode" FullName="Microsoft.Azure.Devices.Client.IotHubClientErrorCode">
  <TypeSignature Language="C#" Value="public enum IotHubClientErrorCode" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi sealed IotHubClientErrorCode extends System.Enum" />
  <TypeSignature Language="DocId" Value="T:Microsoft.Azure.Devices.Client.IotHubClientErrorCode" />
  <TypeSignature Language="VB.NET" Value="Public Enum IotHubClientErrorCode" />
  <TypeSignature Language="F#" Value="type IotHubClientErrorCode = " />
  <AssemblyInfo>
    <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Enum</BaseTypeName>
  </Base>
  <Docs>
    <summary>
            The IoT hub device/module client error code.
            </summary>
    <remarks>To be added.</remarks>
  </Docs>
  <Members>
    <Member MemberName="DeviceMaximumQueueDepthExceeded">
      <MemberSignature Language="C#" Value="DeviceMaximumQueueDepthExceeded" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode DeviceMaximumQueueDepthExceeded = int32(1)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.DeviceMaximumQueueDepthExceeded" />
      <MemberSignature Language="VB.NET" Value="DeviceMaximumQueueDepthExceeded" />
      <MemberSignature Language="F#" Value="DeviceMaximumQueueDepthExceeded = 1" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.DeviceMaximumQueueDepthExceeded" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>1</MemberValue>
      <Docs>
        <summary>
            This actually corresponds to QuotaExceeded. For more information on what causes this error
            and steps to resolve, see <see href="https://docs.microsoft.com/azure/iot-hub/iot-hub-troubleshoot-error-403002-iothubquotaexceeded" />.
            The exception type has not been changed to avoid breaking changes but the inner exception has the correct exception type.
            </summary>
      </Docs>
    </Member>
    <Member MemberName="DeviceMessageLockLost">
      <MemberSignature Language="C#" Value="DeviceMessageLockLost" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode DeviceMessageLockLost = int32(3)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.DeviceMessageLockLost" />
      <MemberSignature Language="VB.NET" Value="DeviceMessageLockLost" />
      <MemberSignature Language="F#" Value="DeviceMessageLockLost = 3" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.DeviceMessageLockLost" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>3</MemberValue>
      <Docs>
        <summary>
            The request failed because attempting to reject/abandon/complete a cloud-to-device message with a lock
            token that has already expired. The lock token expires after the lock timeout set by the service, or if your
            client connection was lost and regained while receiving the message but before you could reject/abandon/complete it.
            </summary>
        <remarks>
            An abandoned message will be re-enqueued in the per-device/module queue, and the <see cref="T:Microsoft.Azure.Devices.Client.IotHubDeviceClient" /> or <see cref="T:Microsoft.Azure.Devices.Client.IotHubModuleClient" /> instance will receive it again.
            A rejected message will be deleted from the queue and not received again by the device.
            For more information on the cause for this error and how to resolve, see <see href="https://docs.microsoft.com/azure/iot-hub/iot-hub-troubleshoot-error-412002-devicemessagelocklost" />.
            For more information on cloud-to-device message lifecycle, see <see href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-c2d#the-cloud-to-device-message-life-cycle" />.
            </remarks>
      </Docs>
    </Member>
    <Member MemberName="DeviceNotFound">
      <MemberSignature Language="C#" Value="DeviceNotFound" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode DeviceNotFound = int32(4)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.DeviceNotFound" />
      <MemberSignature Language="VB.NET" Value="DeviceNotFound" />
      <MemberSignature Language="F#" Value="DeviceNotFound = 4" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.DeviceNotFound" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>4</MemberValue>
      <Docs>
        <summary>
            The request failed because the device is disabled and will be used to set the state to device disabled in the
            connection state handler.
            </summary>
      </Docs>
    </Member>
    <Member MemberName="MessageTooLarge">
      <MemberSignature Language="C#" Value="MessageTooLarge" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode MessageTooLarge = int32(10)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.MessageTooLarge" />
      <MemberSignature Language="VB.NET" Value="MessageTooLarge" />
      <MemberSignature Language="F#" Value="MessageTooLarge = 10" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.MessageTooLarge" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>10</MemberValue>
      <Docs>
        <summary>
            The attempt to send a message fails because the length of the message exceeds the maximum size allowed.
            </summary>
        <remarks>
            When the message is too large for IoT hub you will receive this exception. You should attempt to reduce
            your message size and send again. For more information on message sizes, see
            <see href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-quotas-throttling#other-limits">IoT hub quotas and throttling | Other limits</see></remarks>
      </Docs>
    </Member>
    <Member MemberName="NetworkErrors">
      <MemberSignature Language="C#" Value="NetworkErrors" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode NetworkErrors = int32(5)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.NetworkErrors" />
      <MemberSignature Language="VB.NET" Value="NetworkErrors" />
      <MemberSignature Language="F#" Value="NetworkErrors = 5" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.NetworkErrors" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>5</MemberValue>
      <Docs>
        <summary>
            The attempt to communicate with the IoT hub service fails due to transient network errors after exhausting
            all the retries based on the retry policy set on the client or due to operation timeouts.
            </summary>
        <remark>
            By default, the SDK indefinitely retries dropped connections, unless the retry policy is overridden.
            For more information on the SDK's retry policy and how to override it, see <see href="https://github.com/Azure/azure-iot-sdk-csharp/blob/main/iothub/device/devdoc/retrypolicy.md" />.
            </remark>
      </Docs>
    </Member>
    <Member MemberName="Ok">
      <MemberSignature Language="C#" Value="Ok" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode Ok = int32(0)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.Ok" />
      <MemberSignature Language="VB.NET" Value="Ok" />
      <MemberSignature Language="F#" Value="Ok = 0" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.Ok" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>0</MemberValue>
      <Docs>
        <summary>
            The request completed without exception.
            </summary>
      </Docs>
    </Member>
    <Member MemberName="PreconditionFailed">
      <MemberSignature Language="C#" Value="PreconditionFailed" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode PreconditionFailed = int32(9)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.PreconditionFailed" />
      <MemberSignature Language="VB.NET" Value="PreconditionFailed" />
      <MemberSignature Language="F#" Value="PreconditionFailed = 9" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.PreconditionFailed" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>9</MemberValue>
      <Docs>
        <summary>
            The ETag in the request does not match the ETag of the existing resource.
            </summary>
        <remark>
            The ETag is a mechanism for protecting against the race conditions of multiple clients updating the same resource and overwriting each other.
            </remark>
      </Docs>
    </Member>
    <Member MemberName="QuotaExceeded">
      <MemberSignature Language="C#" Value="QuotaExceeded" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode QuotaExceeded = int32(2)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.QuotaExceeded" />
      <MemberSignature Language="VB.NET" Value="QuotaExceeded" />
      <MemberSignature Language="F#" Value="QuotaExceeded = 2" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.QuotaExceeded" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>2</MemberValue>
      <Docs>
        <summary>
            The request failed because the quota for such operations has been exceeded.
            </summary>
      </Docs>
    </Member>
    <Member MemberName="ServerBusy">
      <MemberSignature Language="C#" Value="ServerBusy" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode ServerBusy = int32(11)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.ServerBusy" />
      <MemberSignature Language="VB.NET" Value="ServerBusy" />
      <MemberSignature Language="F#" Value="ServerBusy = 11" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.ServerBusy" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>11</MemberValue>
      <Docs>
        <summary>
            The request was rejected by the service because it is too busy to handle it right now.
            </summary>
        <remarks>
            This exception typically means the service is unavailable due to high load or an unexpected error and is usually transient.
            The best course of action is to retry your operation after some time.
            By default, the SDK will utilize the <see cref="T:Microsoft.Azure.Devices.Client.IotHubClientExponentialBackoffRetryPolicy" /> retry policy.
            </remarks>
      </Docs>
    </Member>
    <Member MemberName="ServerError">
      <MemberSignature Language="C#" Value="ServerError" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode ServerError = int32(12)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.ServerError" />
      <MemberSignature Language="VB.NET" Value="ServerError" />
      <MemberSignature Language="F#" Value="ServerError = 12" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.ServerError" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>12</MemberValue>
      <Docs>
        <summary>
            The service encountered an error while handling the request.
            </summary>
        <remarks>
            This exception typically means the IoT hub service has encountered an unexpected error and is usually transient.
            Please review the
            <see href="https://docs.microsoft.com/azure/iot-hub/iot-hub-troubleshoot-error-500xxx-internal-errors">500xxx Internal errors</see>
            guide for more information. The best course of action is to retry your operation after some time. By default,
            the SDK will utilize the <see cref="T:Microsoft.Azure.Devices.Client.IotHubClientExponentialBackoffRetryPolicy" /> retry policy.
            </remarks>
      </Docs>
    </Member>
    <Member MemberName="Suspended">
      <MemberSignature Language="C#" Value="Suspended" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode Suspended = int32(6)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.Suspended" />
      <MemberSignature Language="VB.NET" Value="Suspended" />
      <MemberSignature Language="F#" Value="Suspended = 6" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.Suspended" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>6</MemberValue>
      <Docs>
        <summary>
            The IoT hub has been suspended. This is likely due to exceeding Azure spending limits.
            </summary>
        <remark>
            To resolve the error, check the Azure bill and ensure there are enough credits.
            </remark>
      </Docs>
    </Member>
    <Member MemberName="Throttled">
      <MemberSignature Language="C#" Value="Throttled" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode Throttled = int32(8)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.Throttled" />
      <MemberSignature Language="VB.NET" Value="Throttled" />
      <MemberSignature Language="F#" Value="Throttled = 8" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.Throttled" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>8</MemberValue>
      <Docs>
        <summary>
            The request failed because the IoT hub exceed the limits based on the tier of the hub.
            </summary>
        <remark>
            Retrying with exponential back-off could resolve this error. For information on the IoT hub quotas and throttling, see <see href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-quotas-throttling" />.
            </remark>
      </Docs>
    </Member>
    <Member MemberName="Timeout">
      <MemberSignature Language="C#" Value="Timeout" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode Timeout = int32(7)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.Timeout" />
      <MemberSignature Language="VB.NET" Value="Timeout" />
      <MemberSignature Language="F#" Value="Timeout = 7" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.Timeout" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>7</MemberValue>
      <Docs>
        <summary>
            The request failed because the operation timed out. This can be caused by underlying network issues 
            or by the server being too busy to handle the request.
            </summary>
      </Docs>
    </Member>
    <Member MemberName="TlsAuthenticationError">
      <MemberSignature Language="C#" Value="TlsAuthenticationError" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode TlsAuthenticationError = int32(14)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.TlsAuthenticationError" />
      <MemberSignature Language="VB.NET" Value="TlsAuthenticationError" />
      <MemberSignature Language="F#" Value="TlsAuthenticationError = 14" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.TlsAuthenticationError" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>14</MemberValue>
      <Docs>
        <summary>
            The request failed because of TLS authentication error.
            </summary>
        <remarks>
            This error may happen when the remote certificate presented could not be validated, the TLS version is different between client requested
            auth and service minimum requirement, cipher suites to be used could not be agreed upon, etc. The best course of action is to check your
            device certificates and ensure they are up-to-date.
            </remarks>
      </Docs>
    </Member>
    <Member MemberName="Unauthorized">
      <MemberSignature Language="C#" Value="Unauthorized" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype Microsoft.Azure.Devices.Client.IotHubClientErrorCode Unauthorized = int32(13)" />
      <MemberSignature Language="DocId" Value="F:Microsoft.Azure.Devices.Client.IotHubClientErrorCode.Unauthorized" />
      <MemberSignature Language="VB.NET" Value="Unauthorized" />
      <MemberSignature Language="F#" Value="Unauthorized = 13" Usage="Microsoft.Azure.Devices.Client.IotHubClientErrorCode.Unauthorized" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>Microsoft.Azure.Devices.Client.IotHubClientErrorCode</ReturnType>
      </ReturnValue>
      <MemberValue>13</MemberValue>
      <Docs>
        <summary>
            The request failed because the provided credentials are out of date or incorrect.
            </summary>
        <remarks>
            This exception means the client is not authorized to use the specified IoT hub. Please review the
            <see href="https://docs.microsoft.com/azure/iot-hub/iot-hub-troubleshoot-error-401003-iothubunauthorized">401003 IoTHubUnauthorized</see>
            guide for more information.
            </remarks>
      </Docs>
    </Member>
  </Members>
</Type>
